### ISSUE TITLE
`iadmin rmuser` and `iadmin rmgroup` can function for each other

### ISSUE THREAD

**Issue by mstfdkmn** [irods/irods#7380]
- [x] main
- [x] 4-3-stable

---

### iRODS Version, OS and Version
4.3.0, almalinux8

### What did you try to do?
I accidentally called `iadmin rmuser` to remove a group.

### Expected behavior
Not to expect the group to be removed and I expected to see an informative error.

### Observed behavior (including steps to reproduce, if applicable)
The group was removed like a user and I didnt see any error. I see the same behavior for `iadmin rmgroup user`

**Comment by trel** [comment]
groups *are* users in the system/database... but... this still feels like a bug.

**Comment by FifthPotato** [comment]
> groups _are_ users in the system/database... but... this still feels like a bug.

Funny thing about that...

https://github.com/irods/irods_client_icommands/blob/d2c409184ff4d4a59aed41c7674a0d7c992bfefc/src/iadmin.cpp#L1626-L1630

iadmin actually does this on purpose. Indeed...

https://github.com/irods/irods/blob/17422abcce63ccd8ae0ed5a5fe711e15d0456658/server/api/src/rsGeneralAdmin.cpp#L1136

`rsGeneralAdmin` doesn't look like it even has a special block for removing groups.

Does this mean we should add an additional case to `rsGeneralAdmin`?

**Comment by alanking** [comment]
Yes, I think we will need to modify `iadmin` as well as the GeneralAdmin API such that they are separated. It is possible that the database plugin will need to grow a new thing as well, but that remains to be seen, I guess.

**Comment by korydraughn** [comment]
Agreed.

At the very least, `rsGeneralAdmin` should detect whether the entity is a user/group and respond accordingly.

@FifthPotato Please investigate which database PEP is invoked for this operation and what information is passed to it. That should help us determine if the database plugin needs a new operation.

**Comment by trel** [comment]
without looking - i think i would expect the admin endpoint to detect a mismatch and say 'no' to the requesting client.

whether there should be an additional/different mechanism *in* the server feels like a different/separate question.

**Comment by korydraughn** [comment]
That's fair.

**Comment by alanking** [comment]
@FifthPotato  - Please close if complete. Thanks!

**Comment by korydraughn** [comment]
Re-opening.

@FifthPotato Please update the user administration library. Two of the unit tests are failing. The junit output will tell you where the issue is (i.e. avoid passing `--discard-logs` to get the junit output).
